File: Studytidy/app/src/main/java/com/example/stasy/study_tidy/DateStorage.java
package com.example.stasy.study_tidy;

import android.util.Pair;

import java.util.ArrayList;
import java.util.HashMap;

public class DateStorage {
    //дата, направление дела, список дел
    private HashMap<String, HashMap<String, ArrayList<String>>> dataStorage = new HashMap<String, HashMap<String, ArrayList<String>>>();
    private ArrayList<String> events = new ArrayList<String>();
    private HashMap<String, ArrayList<String>> hashMap = new HashMap<String, ArrayList<String>>();

    public DateStorage()
    {

    }
    public void addEvent(String data, String course, String event)
    {
        if (dataStorage.get(data) == null)
        {
            events = new ArrayList<String>();
        }
        events.add(event);
        hashMap.put(course,events);
        dataStorage.put(data, hashMap);

    }
    public void changeEvent()
    {

    }
    public void deleteEvent()
    {

    }
    public ArrayList<String> todayEvent(String date,String key)
    {
        ArrayList<String> todayEvents = new ArrayList<String>();
        todayEvents = dataStorage.get(date).get(key);
        return todayEvents;
    }

}

File: Studytidy/app/src/main/java/com/example/stasy/study_tidy/DateStorage.kt
package com.example.stasy.study_tidy

import android.util.Pair
import com.google.gson.internal.LinkedTreeMap
import java.io.FileWriter
import java.io.Writer

import java.util.ArrayList
import java.util.HashMap

class DateStorage {

    companion object {
        //дата, направление дела, список дел
        val educationEvent = "Учебная"
        val funEvent = "Внеучебная"
        var dataStorage = HashMap<String, LinkedTreeMap<String, ArrayList<String>>>()
        val filename = "myfile.json"

    fun deleteEvent(date: String, key: String, position: Int) {
        if(position < dataStorage.get(date)?.get(key)?.size!!)
            dataStorage.get(date)?.get(key)?.removeAt(position)
    }

    fun todayEvent(date: String, key: String): ArrayList<String> {
        var todayEvents = ArrayList<String>()
        if (dataStorage.containsKey(date) && dataStorage[date]!!.containsKey(key)) {
            todayEvents = dataStorage[date]!!.get(key)!!
        } else {
            todayEvents.add(0, "")

        }
        return todayEvents

    }
    fun addEvent(s: String, s1: String, toString: String) {
        if (!dataStorage.containsKey(s))
            dataStorage[s] = LinkedTreeMap()
        if (!dataStorage[s]!!.containsKey(s1))
            dataStorage[s]!![s1] = ArrayList()
        dataStorage[s]!![s1]!!.add(toString)
    }

    fun setEvent(s: String, s1: String, toString: String, position: Int) {
        if(position < dataStorage.get(s)?.get(s1)?.size!!)
            dataStorage.get(s)?.get(s1)?.set(position, toString)
    }
    }
}
